### PR TITLE
fix problems with builder position

### LINKED ISSUE #1535: builders die building a fancy field
### Minecolonies version

version 0.8.4063

### Expected behavior

- building a fancy field

 ### Actual behaviour

- builders die placing redstone lamp in the 3 deep hole in the middle

### Steps to reproduce the problem

- i had 2 builders die back to back to this.. then finished it myself, dont like to kill more builders building this :)

can clear out the area to have room under it or as the structure diagram is put into the world to avoid this however i reckon most players will put the dirt level at ground level when starting this.

### LINKED ISSUE #1548: Builders standing somewhere they will place blocks, in turn suffocating themselves
### Minecolonies version

version 1.10.2-0.8.4410 or curse 0.8.14

### Expected behavior

- When a builder is building, they will always stand somewhere in which they will not place a block on top of themselves, in turn suffocating in the block.

### Actual behaviour

- During building, the builder will stand in a place in which they are going to place a block, therefore eventually suffocating themselves, I notice this most on a scan, which is simply a 16x16x6 solid placeholder block scan to make filling areas with dirt easier, they will stand on the lower levels of this build and end up suffocating when they build the higher levels.

### Steps to reproduce the problem

1. Make a 16x16x6 scan of solid blocks 
2. Have the builder build it
3. ...
4. Profit

(I'm sure there are other means to reproduce it, but this happens with every scan of this type I place, therefore should work for someone else)

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #1548
Closes #1535

# Changes proposed in this pull request:
- Fix builder dying when building (suffocation)
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java [hunk 1]
```diff
@@ -551,19 +561,19 @@ protected boolean isAlreadyCleared()
...
-        final int z1 = wrapper.getPosition().getZ() - wrapper.getOffset().getZ() - 1;
-        final int x3 = wrapper.getPosition().getX() + (wrapper.getWidth() - wrapper.getOffset().getX());
-        final int z3 = wrapper.getPosition().getZ() + (wrapper.getLength() - wrapper.getOffset().getZ());
+        final int x1 = wrapper.getPosition().getX() - wrapper.getOffset().getX() - STAND_OFFSET;
+        final int z1 = wrapper.getPosition().getZ() - wrapper.getOffset().getZ() - STAND_OFFSET;
+        final int x3 = wrapper.getPosition().getX() + (wrapper.getWidth() - wrapper.getOffset().getX()) + STAND_OFFSET;
+        final int z3 = wrapper.getPosition().getZ() + (wrapper.getLength() - wrapper.getOffset().getZ() + STAND_OFFSET);
 
-        final BlockPos[] edges = new BlockPos[]{new BlockPos(x1, 70, z1), new BlockPos(x3, 70, z1), new BlockPos(x1, 70, z3), new BlockPos(x3, 70, z3)};
+        final BlockPos[] edges = new BlockPos[]{new BlockPos(x1, BASE_Y_HEIGHT, z1), new BlockPos(x3, 70, z1), new BlockPos(x1, BASE_Y_HEIGHT, z3), new BlockPos(x3, BASE_Y_HEIGHT, z3)};
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java]: MAJOR Split this 185 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 35 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 32 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L52](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L52): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L187](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L187): The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requestsystem](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/colony/requestsystem): Add a 'package-info.java' file to document the 'requestsystem' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [init](https://github.com/Minecolonies/minecolonies/blob/eaaa172131ee61cb305c7c6075b67749c93381e0/src/main/java/com/minecolonies/coremod/colony/requestsystem/init): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 34 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 32 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L52](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L52): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 31 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [EntityAIStructureBuilder.java#L187](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java#L187): The Cyclomatic Complexity of this method "requestMaterials" is 17 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [requestsystem](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/colony/requestsystem): Add a 'package-info.java' file to document the 'requestsystem' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [init](https://github.com/Minecolonies/minecolonies/blob/335f5efa07551263e2ada0aa2f803a79d1c8f533/src/main/java/com/minecolonies/coremod/colony/requestsystem/init): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Asherslab** [comment]
:shipit: looks simple and good to me!

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1552/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Review by marchermans — APPROVED** [review]

**Pull request merged**